packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -661,39 +765,175 @@ abstract class TableBase extends Resource implements ITable {
   /**
    * Metric for the system errors this table
    *
-   * @default sum over a minute
+   * @deprecated use `metricSystemErrorsForOperations`.
    */
   public metricSystemErrors(props?: cloudwatch.MetricOptions): cloudwatch.Metric {
+
+    if (!props?.dimensions?.Operation || !props?.dimensions?.TableName) {
+      // 'Operation' must be passed because its an operational metric.
+      // 'TableName' must be passed because the dimensions here will override the default ones which contain the table name.
+      throw new Error("Both 'Operation' and 'TableName' dimensions must be passed for the 'SystemErrors' metric.");
+    }
+
     return this.metric('SystemErrors', { statistic: 'sum', ...props });
   }
 
   /**
-   * Metric for the user errors this table
+   * Metric for the user errors. Note that this metric reports user errors across all
+   * the tables in the account and region the table resides in.
    *
-   * @default sum over a minute
+   * By default, the metric will be calculated as a sum over a period of 5 minutes.
+   * You can customize this by using the `statistic` and `period` properties.
    */
   public metricUserErrors(props?: cloudwatch.MetricOptions): cloudwatch.Metric {
-    return this.metric('UserErrors', { statistic: 'sum', ...props });
+
+    if (props?.dimensions) {
+      throw new Error("'dimensions' is not supported for the 'UserErrors' metric");
+    }
+
+    // overriding 'dimensions' here because this metric is an account metric.
```

Comment:
If this metric does not depend on the table object, it should have been a `static` method.

Reply:
yes, but that would be breaking.

Reply:
Which is we would do add new method/`@deprecated`.